qtdialogs/DlgSetupManager.py

```diff
@@ -0,0 +1,1303 @@
+         else:
+            def stageCheckboxHandler(state):
+               self.onStageCheckboxChanged(walletId, state == QtCore.Qt.Checked)
+            cell, cb = makeCheckboxCell(True, True, stageCheckboxHandler)
+
+            self.walletList.setItemWidget(item, 0, cell)
+            self.walletIdToCheckbox[walletId] = cb
+
+         # Persist metadata on the row for later handlers
+         stagedFlag = bool(entry.staged)
+         item.setData(0, QtCore.Qt.UserRole, {
+            'walletId': walletId,
+            'path': walletPath,
+            'staged': stagedFlag
+         })
+
+         # Column 4: Action buttons (left-aligned for better appearance)
+         if isLegacy:
+            def migrateHandler(_):
+               self.migrateWallet(walletPath)
+            self._addActionButton(item, self.tr('Migrate'), migrateHandler)
+         elif isEncrypted:
+            def unlockHandler(_):
+               self.unlockWallet(walletId)
+            self._addActionButton(item, self.tr('Unlock'), unlockHandler)
+         else:
+            # Render nothing in Action column for ready wallets
+            pass
+
+         return True
+      except Exception:
+         return False
+
+   def _walletStateLabel(self, entry, walletId):
+      """Map bridge state to UI label and flags.
+
+      Returns (label, isLegacy, isEncrypted). Avoids per-row imports by using
+      the bridge-provided enum: unknown(0), legacy(1), migrated(2),
+      encrypted(3), ready(4), loaded(5).
+      """
+      label = self.tr('Ready')
+      isLegacy = False
+      isEncrypted = False
+
+      try:
+         stateVal = entry.state
+         # Handle numeric enums and string fallbacks defensively
+         if isinstance(stateVal, int):
+            if stateVal == WALLET_STATE_LEGACY:
+               label = self.tr('Legacy')
+               isLegacy = True
+            elif stateVal == WALLET_STATE_ENCRYPTED:
+               label = self.tr('Encrypted')
+               isEncrypted = True
+            elif stateVal == WALLET_STATE_LOADED:
+               # treat loaded as ready for this setup view
+               label = self.tr('Ready')
+            elif stateVal in (WALLET_STATE_MIGRATED, 4):
+               label = self.tr('Ready')
+            elif stateVal == WALLET_STATE_UNKNOWN:
+               label = self.tr('Unknown')
+         else:
+            s = str(stateVal).lower()
+            if s == 'legacy':
+               label = self.tr('Legacy')
+               isLegacy = True
+            elif s == 'encrypted':
+               label = self.tr('Encrypted')
+               isEncrypted = True
+            elif s == 'loaded':
+               label = self.tr('Ready')
+            elif s in ('migrated', 'ready'):
+               label = self.tr('Ready')
+            elif s == 'unknown':
+               label = self.tr('Unknown')
+      except Exception:
+         # Default to Ready on any unexpected schema
+         label = self.tr('Ready')
```

Comment:
1. what can raise here? why the try block?
2. default handling is to consider the wallet ready? on what basis?